src/Elliptic/Systems/Poisson/Equations.cpp

```diff
@@ -27,12 +28,17 @@ template <size_t Dim>
 void non_euclidean_fluxes(
     const gsl::not_null<tnsr::I<DataVector, Dim>*> flux_for_field,
     const tnsr::II<DataVector, Dim>& inv_spatial_metric,
-    const Scalar<DataVector>& det_spatial_metric,
     const tnsr::i<DataVector, Dim>& field_gradient) noexcept {
   raise_or_lower_index(flux_for_field, field_gradient, inv_spatial_metric);
-  for (size_t i = 0; i < Dim; i++) {
-    flux_for_field->get(i) *= sqrt(get(det_spatial_metric));
-  }
+}
+
+template <size_t Dim>
+void add_non_euclidean_sources(
+    const gsl::not_null<Scalar<DataVector>*> source_for_field,
+    const tnsr::i<DataVector, Dim>& christoffel_contracted,
+    const tnsr::I<DataVector, Dim>& flux_for_field) noexcept {
+  get(*source_for_field) -=
```

Comment:
Our tensors have two kinds of `get` functions: (1) Member functions like `tensor.get(i, j)` for runtime-access to the components and (2) free functions like `get<0, 1>(tensor)` for when the components are known at compile-time. Since a `Scalar` only has a single component we can always use the compile-time version `get(scalar)`.